### PR TITLE
Add support for raw-dylib

### PR DESCRIPTION
**Pull request by dpaoliello**
This adds support for generating MSVC compatible import libraries for the `raw-dylib` feature and removes the patches for the standard library that worked around the lack of support.

Also, kudos to @simonbuchan who created an [earlier prototype of `raw-dylib` support](https://github.com/rust-lang/rustc_codegen_cranelift/pull/1414) as I used that as a guide for parts of this implementation.

### CODE REVIEW

**Comment by bjorn3** [comment]
I've updated rustc on the master branch. Would you mind rebasing this PR?

**Comment by dpaoliello** [comment]
> I've updated rustc on the master branch. Would you mind rebasing this PR?

Done

**Comment by bjorn3** [comment]
Thanks a lot for all the work implementing this.

File: Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ cranelift-object = { version = "0.110.1" }
 target-lexicon = "0.12.0"
 gimli = { version = "0.28", default-features = false, features = ["write"]}
 object = { version = "0.36", default-features = false, features = ["std", "read_core", "write", "archive", "coff", "elf", "macho", "pe"] }
+ar_archive_writer = "0.3"
```
Comment (bjorn3) [inline Cargo.toml:20]: I just noticed this dependency wasn't removed. Will do so myself.

Comment (dpaoliello) [reply Cargo.toml:20]: It's still needed: we call into `ar_archive_writer` to create the actual import library

Comment (bjorn3) [reply Cargo.toml:20]: The sysroot already contains a copy of ar_archive_writer for use by cg_ssa. cg_clif can use this copy: https://github.com/rust-lang/rust/pull/128896/commits/6e5873c08c13b7aebae6e277caa39177e6c2b5e3

**Comment by bjorn3** [comment]
https://github.com/rust-lang/rust/pull/128896 will update the cg_clif shipped with rustup.

**Pull request merged**